packages/shared/src/components/sidebar/Sidebar.tsx

```diff
@@ -53,7 +47,8 @@ export default function Sidebar({
     loadedSettings,
     optOutWeeklyGoal,
   } = useContext(SettingsContext);
-  const [showSettings, setShowSettings] = useState(false);
+  const { modal, openModal } = useLazyModal();
+  const showSettings = modal?.type === LazyModal.UserSettings;
```

Comment:
Why do you need this here?

Reply:
Oh ok, just didn't really get why we needed it.
So what changes based on this state down the props?

Reply:
We disable showing the new rank popup while the modal is active: https://github.com/dailydotdev/apps/blob/mi-131-profile-menu/packages/shared/src/components/sidebar/SidebarBottomSection.tsx#L60

And we show the sidebar `Customize` menu as active: https://github.com/dailydotdev/apps/blob/mi-131-profile-menu/packages/shared/src/components/sidebar/ManageSection.tsx/#L51-L53